pysph/sph/bc/inlet_outlet_manager.py

```diff
@@ -0,0 +1,318 @@
+            dx = info.dx
+            if (info.pa_name == pa.name):
+                x = pa.x
+                y = pa.y
+                z = pa.z
+                xmax, xmin = max(x)+dx/2, min(x)-dx/2
+                ymax, ymin = max(y)+dx/2, min(y)-dx/2
+                zmax, zmin = max(z)+dx/2, min(z)-dx/2
+                xdist = xmax - xmin
+                ydist = ymax - ymin
+                zdist = zmax - zmin
+                xn, yn, zn = info.normal[0], info.normal[1], info.normal[2]
+                info.length = abs(xdist*xn+ydist*yn+zdist*zn)
+                info.bound = [[xmin, ymin, zmin], [xmax, ymax, zmax]]
+
+    def update_inlet_outlet_domain(self):
+        """Updates the domain information"""
+        x = []
+        y = []
+        z = []
+        all_info = self.inletinfo + self.outletinfo
+        for info in all_info:
+            x.append(info.bound[0][0])
+            x.append(info.bound[1][0])
+            y.append(info.bound[0][1])
+            y.append(info.bound[1][1])
+            z.append(info.bound[0][2])
+            z.append(info.bound[1][2])
+        xmax, xmin = max(x), min(x)
+        ymax, ymin = max(y), min(y)
+        zmax, zmin = max(z), min(z)
+
+        for info in all_info:
+            info.domain = [[xmin, ymin, zmin], [xmax, ymax, zmax]]
+
+    def add_io_properties(self, pa, scheme=None):
+        """Add properties to be used in inlet/outlet equations
+
+        Parameters
+        ----------
+
+        pa : particle_array
+            Particle array of inlet/outlet
+        scheme : pysph.sph.scheme
+            The insance of scheme class
+        """
+        pass
+
+    def get_io_names(self):
+        """return all the names of inlets and outlets"""
+        return self.inlets + self.outlets
+
+    def get_stepper(self, scheme, integrator, **kw):
+        """Returns the steppers for inlet/outlet
+
+        Parameters
+        ----------
+
+        scheme : pysph.sph.scheme
+            The instance of the scheme class
+        intergrator : pysph.sph.integrator
+            The parent class of the integrator
+        **kw : extra arguments
+            Extra arguments depending upon the scheme used
+        """
+        raise NotImplementedError()
+
+    def setup_iom(self, dim, kernel):
+        """Essential data passed
+
+        Parameters
+        ----------
+
+        dim : int
+            dimension of the problem
+        kernel : pysph.base.kernel
+            the kernel instance
+        """
+        self.dim = dim
+        self.kernel = kernel
+
+    def get_equations(self, scheme, **kw):
+        """Returns the equations for inlet/outlet
+
+        Parameters
+        ----------
+
+        scheme : pysph.sph.scheme
+            The instance of the scheme class
+        **kw : extra arguments
+            Extra arguments depending upon the scheme used
+        """
+        raise NotImplementedError()
+
+    def get_equations_post_compute_acceleration(self, scheme, **kw):
```

Comment:
This doesn't seem to be used or do you have other applications that need this?  It still seems a little weird.

Reply:
Not using it, but will use later. but for now removing it.